repos/system_upgrade/el7toel8/actors/satelliteupgradefacts/actor.py

```diff
@@ -0,0 +1,41 @@
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.models import InstalledRPM, Module, RepositoriesSetupTasks, RpmTransactionTasks
+from leapp.tags import IPUWorkflowTag, FactsPhaseTag
+
+
+class SatelliteUpgradeFacts(Actor):
+    """
+    No documentation has been provided for the satellite_upgrade_facts actor.
+    """
+
+    name = 'satellite_upgrade_facts'
+    consumes = (InstalledRPM, )
+    produces = (RepositoriesSetupTasks, RpmTransactionTasks)
+    tags = (IPUWorkflowTag, FactsPhaseTag)
+
+    def process(self):
+        local_postgresql = has_package(InstalledRPM, 'rh-postgresql12-postgresql-server')
+        # remove packages we don't need anymore
+        self.produce(RpmTransactionTasks(
+            to_remove=['tfm-runtime', 'rh-redis5-runtime', 'rh-ruby27-runtime'],
+            # you'll need to define the repos in leapp_upgrade_repositories.repo
+            # until this is on the CDN
+            to_install=['rubygem-foreman_maintain', 'foreman-installer-katello'],
+            modules_to_enable=[Module(name='ruby', stream='2.7')]
+            )
+        )
+        if local_postgresql:
+            self.produce(RpmTransactionTasks(
+                modules_to_enable=[Module(name='postgresql', stream='12')],
+                to_remove=['rh-postgresql12-runtime'],
+                to_install=['postgresql', 'postgresql-server']
```

Comment:
In some environments we've also seen `-contrib` installed. Would it be lost after this?